diff --git a/fs/ext4/auto_defrag.c b/fs/ext4/auto_defrag.c
index 01ca855..1753627 100644
--- a/fs/ext4/auto_defrag.c
+++ b/fs/ext4/auto_defrag.c
@@ -133,6 +133,7 @@ static void ext4_read_and_dirty_end_io(struct bio *bio, int err) {
 		if (uptodate) {
 			SetPageUptodate(page);
 			set_page_buffers_dirty(page);
+			ext4_remap_blocks(page->mapping->host, page->index, 1);
 		} else {
 			ClearPageUptodate(page);
 			SetPageError(page);
